### PR TITLE
Add feature to filter files/directories

### PR DESCRIPTION
**Pull request by epcim**
```

[diff]
  color = auto
  tool = icdiff

[difftool "icdiff"]
      cmd = /usr/local/bin/icdiff -r -IF '.*(pbgo|pbdoc|pbswagger|pbcpp|lock)' --strip-trailing-cr --line-numbers $LOCAL $REMOTE


```

### CODE REVIEW

File: icdiff [hunk 1]
```diff
@@ -480,6 +480,16 @@ def create_option_parser():
...
+    parser.add_option("-F", "--filepath-regex",
+                      action="store",
+                      default=None,
+                      type="string",
+                      dest='filepath_regex',
+                      help="Process files (full path) that match this regex pattern")
+    parser.add_option("-I", "--inverse-filepath-regex", default=False,
+                      action="store_true",
+                      help="Inverse regex used in --filepath-regex used to filter"
+                      "files/directories")
```
Comment (jeffkaufman) [inline icdiff]: For consistency with `--exclude-lines` which also takes a regex, what would you think of `--exclude-path` and `--include-path`?

The idea would be:

* If `--include-path` is specified, then only paths matching that regexp are included
* If `--exclude-path` is specified, then all paths matching that regexp are not included

I might be confused, but I can't come up with a case where the order of applying `include` and `exclude` matters.

Comment (jeffkaufman) [reply icdiff]: Optionally, you could add `--include-lines` to diff only lines matching the regexp, just for even more consistency.

**Comment by tflhyl** [comment]
hi is there any plan to continue this?

**Comment by jeffkaufman** [comment]
@epcim are you thinking of picking this back up?

@tflhyl you'e welcome to work on this if you want!

**Comment by epcim** [comment]
I missed the commets will have a look to implement suggested.

**Comment by jeffkaufman** [comment]
It doesn't look like this is active anymore, so closing. If you or anyone else wants to pick this back up in the future, that's fine!

**Pull request closed without merging**